### DISCUSSION TITLE
What's up with the Window properties added in 0.10.x?

### DISCUSSION THREAD

**Discussion by Splitwirez** [AvaloniaUI/Avalonia#6335] [Q&A]
As stated in the title, a number of new properties were added to `Window` as of Avalonia 0.10.x. While their names provide vague hints regarding their purpose, I've found many of them...unclear in their exact function, as follows:

#### `ExtendClientAreaToDecorationsHint`
*When* exactly is this property meant to have...any visible effect at all?

#### `ExtendClientAreaChromeHints`
This property seems to be claiming to do at least two things at once (why?), yet doesn't seem to actually have the described effect...

#### `ExtendClientAreaTitleBarHeightHint`
*When* exactly is this property meant to have...any visible effect at all?

#### `WindowDecorationMargin` and `OffScreenMargin`
What even *are* these properties?

Given prior discussion with @danwalmsley on this matter a while back, I'm left to suspect that the Windows implementations for a lot of this stuff were somewhat built with Windows 10 in mind, and can only infer from there that prior versions of Windows were largely ignored...if this is true, then it would be nice to know *exactly* how these properties are *meant* to behave under ideal circumstances, so I can plan accordingly, and *maaaaaybe* even take a crack at improving the situation for older versions of Windows at some point (provided this interpretation is correct)...

**Comment by kekekeks** [comment]
> a lot of this stuff were somewhat built with Windows 10 in mind

Those are supported on macOS and Windows 10. For Linux we'll probably add an API to forcefully enable our managed window frame, but before that I need to write a huge article why it's a terrible idea if you are planing to target anything other than GNOME.

>What even are these properties? 

They do have documentation in XML comments

>ExtendClientAreaToDecorationsHint
```xml
<summary>
    Gets or sets if the ClientArea is Extended into the Window Decorations (chrome or border).
</summary>
```

If you set this to true, your content will be placed on top of the window frame with the actual top-left corner being (0, 0) coordinate. `WindowDecorationMargin` returns the margins that are added by the window frame:

```xml
<summary>
  Gets the WindowDecorationMargin.
  This tells you the thickness around the window that is used by borders and the titlebar.
</summary>

OffscreenMargin is needed to account for the [old hack that' been there since Windows 3.0](https://devblogs.microsoft.com/oldnewthing/20120326-00/?p=8003).
```xml
<summary>
  Gets the window margin that is hidden off the screen area.
  This is generally only the case on Windows when in Maximized where the window border
  is hidden off the screen. This Margin may be used to ensure user content doesnt overlap this space.
```

`ExtendClientAreaTitleBarHeightHint` allows you to resize window header when that's supported, currently only takes effect on macOS
```xml
<summary>
    Gets or Sets the TitlebarHeightHint for when the client area is extended.
    A value of -1 will cause the titlebar to be auto sized to the OS default.
    Any other positive value will cause the titlebar to assume that height.
</summary>
```
